File: Olympiad/COCI/official/2008/contest3_solutions/cetiri.pas
program cetiri;

function min3( a, b, c : longint ) : longint;
begin
   if a < b then
      if a < c then min3 :=a
      else min3 := c
   else
      if b < c then min3 := b
      else min3 := c;
end;

var
   a1, a2, a3, a4, d : longint;

begin
   readln( a1, a2, a3 );

   a4 := min3( a1, a2, a3 );
   d := min3( abs(a1-a2), abs(a1-a3), abs(a2-a3) );

   while (a4 = a1) or (a4 = a2) or (a4 = a3) do a4 := a4 + d;

   writeln( a4 );
end.

File: Olympiad/COCI/official/2008/contest3_solutions/cetiri.cpp
#include <cstdio>

int abs( int a ) { 
   if( a < 0 ) return -a;
   return a;
}

int min3( int a, int b, int c ) {
   if( a < b )
      if( a < c ) return a;
      else return c;
   else 
      if( b < c ) return b;
      else return c;
}

int main( void ) {
   int a1, a2, a3;
   scanf( "%d%d%d", &a1, &a2, &a3 );
   
   int a4 = min3( a1, a2, a3 );
   int d = min3( abs(a1-a2), abs(a1-a3), abs(a2-a3) );

   while( a4 == a1 || a4 == a2 || a4 == a3 ) a4 += d;

   printf( "%d\n", a4 );
   
   return 0;
}
